### DISCUSSION TITLE
New Send Notes to Promp

### DISCUSSION THREAD

**Discussion by FamedBear16** [logancyang/obsidian-copilot#316] [General]
Can anybody tell me how this works?
I am sending all my Zettelkasten notes to the chat mode.
Does this model use embeddings to cut down on the amount of data sent to Open AI? 
Or will it send to Open AI the text of all my notes? 
I gave it a try and with  GPT-4 Turbo I get this error: "LangChain Error: limit exceeded"
I am sure all my notes are less than 128k tokens.

Would be possibel in the future to say something like
For all my notes in @Cards can you answer this question? (So that only the embeddings will be sent?)

Thanks

**Comment by logancyang** [comment]
Hi @FamedBear16, Chat mode doesn't use embeddings, it plugs the notes directly into the prompt. Not sure what you mean by `sending all my Zettelkasten notes to the chat mode` and how you do that. From your error it seems you indeed passed too much to the model.

QA mode is RAG with embeddings. 

In the future, it will be possible to reference note titles directly in chat just with [[note title]]. With that, there will be 3 ways to reference notes in the Copilot plugin

1. "send notes to prompt" with or without setting the context via a command
2. User custom prompt templating
3. Directly mention the note title in chat

Here's a video I made showing how to use custom prompt templating https://youtu.be/VPNlXeCsH74?si=sp3-qmGdSR493wHC, at the very end I also touched on the "send notes to prompt" button way.

**Comment by FamedBear16** [comment]
HI @logancyang,

This is my use case; I hope you can help me to see if copilot can help.

I have a folder with many notes, this is my "ZKBox" Folder. 
in "ZKBox" there are some notes tagged as #excalidraw that should be excluded from the prompting or RAG indexing.


I would like Copilot to index all my ZKBox notes, and then I would like to QA of my notes with a RAG model
Will this function be available? The ability to  QA for multiple folders and filtering out some files with a tag?


I have tried the Chat mode which does not use a RAG model.
I selected the ZKBox and I think I went over the 124k limits because I have some excalidraw drawings.
Is there any way to filter out files with an #excalidraw tag when passing the context?

It would be helpful if you could add a filter that excludes notes with a given tag when creating a context or a custom prompt.


How would you suggest to use copilot to query my ZKBox?

Many thanks

Pierpaolo

**Comment by logancyang** [comment]
hi @FamedBear16, thanks for the details!

Vault QA (BETA) mode v2.5.0 is coming soon, I think it will be most relevant for you use case. The first iteration will index the whole vault without Exclude filters. Exclude filters for Chat mode context and Vault QA mode context can be added for sure.

I will need people to test the new Vault QA mode and provide feedback. Since this is completely client-side, I do not have access to individual user's search logs, so it will be hard for me to help debug individual cases. So users like you who can share feedback with me will be crucial for future iterations. Stay tuned!

**Reply by FamedBear16** [reply]
HI @logancyang ,

I have just realized that the new functions are available. 
I gave a quick try and it is working fine. Indexing my entire vault of around 3Mln tokens was fast and cheap with Open AI (around 0.06$). 
Great work! Thanks.

As an additional option it would be great if I could filter the notes I am querying
Something like: Based on my note in this /folder/ what is...?

Ideally, it would be nice to set up a series of query filters based on folders and tags.

For example: 
Filter 1: path:myfolder OR path:anotherfolder
Filter 1: tags:#test AND path:folder

have many small notes, so I believe that for me sources set to 3 is way too small.
